exec.go

```diff
@@ -0,0 +1,42 @@
+package gocql
+
+import (
+	"context"
+	"fmt"
+)
+
+// ExecQueryOnHost creates a one-shot control connection to the provided host,
+// and executes a query.
+// Query result is processed by a user provided handler.
+func ExecQueryOnHost(ctx context.Context, addr string, cfg *ClusterConfig, handler func(iter *Iter), statement string, values ...interface{}) error {
+	// Create uninitialised session
+	noInitCfg := *cfg
+	noInitCfg.disableInit = true
+	session, err := NewSession(noInitCfg)
+	if err != nil {
+		return fmt.Errorf("new session: %w", err)
+	}
+	defer session.Close()
+
+	// Get HostInfo for address
+	hosts, err := addrsToHosts([]string{addr}, cfg.Port, cfg.Logger)
+	if err != nil {
+		return fmt.Errorf("addr to host: %w", err)
+	}
+
+	// Create control connection
+	c := createControlConn(session)
+	defer c.close()
+	conn, err := session.connect(ctx, hosts[0], c)
```

Comment:
I looked at the session initialization code and see that it tries to detect the protocol version ([link](https://github.com/scylladb/gocql/blob/35bdbf71260497c5aef7c2d35d02ca6d5c94398f/session.go#L205)) which is not done here. By default, the `ClusterConfig` object uses `0` and detects the version before connecting ([link to `NewCluster`](https://github.com/scylladb/gocql/blob/35bdbf71260497c5aef7c2d35d02ca6d5c94398f/cluster.go#L184-L204), it does not set `ProtoVersion`). 

Looking at `createCluster()`, it seems that it takes the protocol version from a commandline flag. The `integration.sh` script seems to pass version 4 here. Could you check if it works when `ProtoVersion` is set to 0? (I suspect not)

Reply:
That is correct, this is done on purpose, I'll return an error if protocol version is not set.

Reply:
IMO we should either add a comment that ProtoVersion must be set, or set it to 4 if it was not set.

Reply:
Either behavior (error if unset / set to 4 by default) is fine by me. It needs to be documented, though - you usually don't need to set the protocol version when connecting with a full session, so the need to set the protocol version might be a bit surprising if it is not documented.